react/w-books/src/app/components/Routes/components/PublicRoute/index.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { Component, FunctionComponent, ReactNode } from 'react'
+import { Route, Redirect } from 'react-router-dom';
+
+import useSession from '../../../../../hooks/useSession';
+
+import PATHS from '../../paths';
+
+interface PublicRouteProps {
+  component: React.ElementType;
+  restricted?: boolean;
+  path: string;
+  exact: boolean;
+}
+
+export default function PublicRoute({ component: Component, restricted, path, exact }: PublicRouteProps) {
```

Comment:
i think that you can create an only component that validates if is the public and private route.